src/shared/nt-results/ResultsByLoad.react.js

```diff
@@ -0,0 +1,194 @@
+import * as d3 from 'd3'
+import gql from 'graphql-tag'
+import _ from 'lodash'
+import React from 'react'
+import { graphql } from 'react-apollo'
+import { compose } from 'recompose'
+
+import styles from './ResultsByLoad.styl'
+
+const SESSIONS_QUERY = gql`
+  query querySessions ($sessionTypeId: String!) {
+    sessions (sessionTypeId: $sessionTypeId) {
+      actions {
+        actionTypeId,
+        type,
+      }
+    }
+  }
+`
+
+const enhance = compose(
+  graphql(SESSIONS_QUERY, {
+    options: ({ sessionTypeId }) => ({ variables: { sessionTypeId: sessionTypeId } })
+  })
+)
+
+export const mapActionsToNodes = ({ x, y }) => session => session.actions.map(action => {
+  const node = { id: action.actionTypeId, type: action.type, x: x, y: y }
+  return action.type === 'load' ? { ...node, fx: 100, fy: 300 } : node
+})
+
+export const mapActionsToLinks = session => {
+  const actions = session.actions
+  const links = []
+
+  if (actions.length > 1) {
+    for (let i = 0; i < actions.length - 1; i++) {
+      // TODO: In the near future, please do the line for target is source
+      if (actions[i].actionTypeId !== actions[i + 1].actionTypeId) {
+        links.push({
+          source: actions[i].actionTypeId,
+          target: actions[i + 1].actionTypeId
+        })
+      }
+    }
+  }
+  return links
```

Comment:
I don't think it is necessary to change. Reduce make it hard to understand the code while simple for loop is more straightforward.